apps/src/util/CachedElement.jsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
```

Comment:
This component ended up being fairly simple but I like how it separates out this functionality.

Reply:
me too. it's also easy to imagine how this could be updated to provide a default implementation of key generation in the future.